tests/test_submission.py

```diff
@@ -227,6 +227,16 @@ def test_sticky_unsticky(self):
         self.assertFalse(submission_a.refresh().stickied)
         self.assertFalse(submission_b.refresh().stickied)
 
+    @betamax()
+    def test_equality_and_hash(self):
+        subreddit = self.r.get_subreddit(self.sr)
+        submission = list(subreddit.get_hot())[0]
+        same_submission = self.r.get_submission(submission_id=submission.id)
+
+        self.assertNotEqual(id(submission), id(same_submission))
+        self.assertEqual(hash(submission), hash(same_submission))
+        self.assertEqual(submission, same_submission)
+
```

Comment:
Can you also add a case where two instances aren't equal that shouldn't be equal so that we don't accidentally add that regression in the future?